File: 7kue/Homogenous arrays/solve-1.py
def filter_homogenous(arrays):
  new_arrays = []
  for arr in arrays:
    if len(arr) == 0:
      continue
    first_item = arr[0]
    homogenous = True
    for item in arr:
      if type(item) != type(first_item):
        homogenous = False
      
    if homogenous:
      new_arrays.append(arr)
      
  return new_arrays



# Пример использования
arrays = [[1, 5, 4], ["a", 3, 5], ["b"], [], ["1", 2, 3]]
print(filter_homogenous(arrays))  # Вывод: [[1, 5, 4], ['b']]

File: 7kue/Magic Index/solve-1.js
function findMagic(arr) {
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] === i) {
      return i;
    }
  }
  return -1;
}

console.log(findMagic([0, 1, 2, 3, 4])); // 0
console.log(findMagic([1, 2, 3, 4, 5])); // -1
console.log(findMagic([1, 2, 3, 3, 5])); // 3